grpc-gcp/src/main/java/com/google/grpc/gcp/GcpManagedChannelOptions.java

```diff
@@ -244,6 +266,36 @@ public Builder enableNotReadyFallback() {
         notReadyFallbackEnabled = true;
         return this;
       }
+
+      /**
+       * Enable unresponsive connection detection.
+       *
+       * <p>If an RPC channel fails to receive any RPC message from the server for {@code ms}
+       * milliseconds and there were {@code numDroppedRequests} calls (started after the last
```

Comment:
Why do we need `numDroppedRequests`?

Reply:
This is a good question. If we just look for a period of time when we didn't receive anything from the server it could be just an idle connection with no RPCs flowing. If we add a check that there should be at least one request and we will have one deadline_exceeded request - it could be just a single request that the back end failed to respond in time and the connection is actually alive. 
It's ok if we don't reconnect the channel when there is not much traffic - this wouldn't save many requests.
So we want to reduce the false positive detections by ensuring that substantial amount of requests were failing. I. e. we want to let the user choose since what QPS this feature should engage within the specified time period.
For example, configuring this feature with 5 seconds, 10 failed requests means that we will detect an unresponsive connection in 5 seconds if we have >= 2 QPS on the channel, if we have less than 2 QPS we will detect an unresponsive connection in more than 5 seconds - as long as 10 requests will be there.
This parameter is a tradeoff between more false positives and longer unresponsiveness detection in the low-QPS scenario and we want user to decide what's more important for their use-case.

Reply:
Thanks for your explanation